cloudwatch/vespa_cloudwatch_emitter.py

```diff
@@ -0,0 +1,158 @@
+
+            metric_data.extend(self._metric_data_for_node_node(nodes_elem))
+            metric_data.extend(self._metric_data_for_node_services(nodes_elem))
+        return metric_data
+
+    def _metric_data_for_node_node(self, nodes_elem):
+        """
+        Returns a list of MetricsData for the 'node' (= node metrics) element
+        for a Vespa node (here represented by nodes_elem).
+        """
+        if 'node' not in nodes_elem:
+            log.info("No node metrics for node {} (this is expected for self-hosted Vespa)."
+                         .format(nodes_elem.get('hostname')))
+            return []
+        return self._metric_data_for_service_or_node(nodes_elem['node'])
+
+    def _metric_data_for_node_services(self, nodes_elem):
+        """
+        Returns a list of MetricsData for all elements in the 'services'
+        list for a Vespa node (here represented by nodes_elem).
+        """
+        if 'services' not in nodes_elem:
+            log.warning("No services for node {}".format(nodes_elem.get('hostname')))
+            return []
+
+        metric_data = []
+        for service in nodes_elem['services']:
+            metric_data.extend(self._metric_data_for_service_or_node(service))
+        return metric_data
+
+    def _metric_data_for_service_or_node(self, service_or_node):
+        """
+        Returns a list of MetricsData for all elements in the 'metrics' json
+        list of a 'nodes' element's 'node' element or one of the elements
+        in a node's 'services' list.
+        """
+        if 'metrics' not in service_or_node:
+            return []
+        metric_data = []
+        for metrics_elem in service_or_node['metrics']:
+            metric_data.extend(self._get_metrics_with_dimensions(metrics_elem))
+        return metric_data
+
+    def _get_metrics_with_dimensions(self, metrics_elem):
+        """
+        Returns a list of MetricsData for one element in a 'metrics' json list.
+        """
+        if 'values' not in metrics_elem:
+            return []
+        metric_data = []
+        dimensions = self._get_dimensions(metrics_elem)
+        for name, value in metrics_elem['values'].items():
+            metric_data.append({'MetricName': name,
+                                'Value': value,
+                                'Unit': 'None',
+                                'Dimensions': dimensions})
+        return metric_data
+
+    def _get_dimensions(self, metrics):
+        dimensions = []
+        if 'dimensions' in metrics:
+            dimensions_json = metrics['dimensions']
+            for dim, dim_val in dimensions_json.items():
+                dimensions.append({'Name': dim,
+                                   'Value': dim_val})
+        return dimensions
+
+    def split_list(self, lst, chunk_size):
+        """
+        Splits the given list into chunks and returns the result as a list of lists/chunks
+        """
+        return list(self._chunk(lst, chunk_size))
```

Comment:
Could do list comprehension directly, e.g.
`return [lst[i:i + chunk_size] for i in range(0, len(lst), chunk_size)]`